source/fuzz/transformation_add_local_variable.cpp

```diff
@@ -70,11 +70,12 @@ bool TransformationAddLocalVariable::IsApplicable(
 void TransformationAddLocalVariable::Apply(
     opt::IRContext* ir_context,
     TransformationContext* transformation_context) const {
-  fuzzerutil::AddLocalVariable(ir_context, message_.fresh_id(),
-                               message_.type_id(), message_.function_id(),
-                               message_.initializer_id());
+  opt::Instruction* new_instruction = fuzzerutil::AddLocalVariable(
+      ir_context, message_.fresh_id(), message_.type_id(),
+      message_.function_id(), message_.initializer_id());
 
-  ir_context->InvalidateAnalysesExceptFor(opt::IRContext::kAnalysisNone);
+  // Inform the def-use manager about the new instruction.
+  ir_context->get_def_use_mgr()->AnalyzeInstDefUse(new_instruction);
```

Comment:
Not sure if we need to set a basic block for the newly added instruction (`set_basic_block` method).

Reply:
You're right - and actually it turns out this was making the long running fuzzer tests fail. (I'm hoping that when we're done with improving the efficiency of the fuzzer they will run a lot more quickly so that I'll be more disciplined about running them locally).

Fixed, and I also augmented the tests for all of the modified transformations with some assertions to check that the necessary updates to relevant analyses happen. I'll try to go back and do that for the other transformations I updated recently.